### PR TITLE
add flag to control cluster telemetry metadata

### PR DESCRIPTION
**Pull request by ramaraochavali**
This is not a huge perf impact but with many clusters (especially in large k8s clusters) this is adding some overhead. So adding a flag to control this behaviour. Ideally we can deprecate existing `PILOT_ENDPOINT_TELEMETRY_LABEL` and remove it later. For now I have added a new flag that works for both.

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/features/pilot.go [hunk 1]
```diff
@@ -400,6 +400,10 @@ var (
 			"Setting the timeout to 0 disables this behavior.",
 	).Get()
 
+	EnableTelemetry = env.RegisterBoolVar("PILOT_ENABLE_TELEMETRY", true,
```
Comment (bianpengyuan) [inline pilot/pkg/features/pilot.go:403]: Maybe `ENABLE_TELEMETRY_LABEL`? `ENABLE_TELEMETRY` sound ambiguous to me.

**Review by bianpengyuan — COMMENTED** [review]
Disable cluster label will make your server side metrics missing destination service related information. Are you sure you would like to disable this?

**Comment by ramaraochavali** [comment]
> Disable cluster label will make your server side metrics missing destination service related information. Are you sure you would like to disable this?

Yes. We do not use telemetry at all. We use Envoy stats for our telemetry

File: pilot/pkg/features/pilot.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/features/pilot.go]: Ok. Makes sense. Changed.

File: pilot/pkg/features/pilot.go [hunk 2]
```diff
@@ -400,7 +400,7 @@ var (
 			"Setting the timeout to 0 disables this behavior.",
 	).Get()
 
-	EnableTelemetry = env.RegisterBoolVar("PILOT_ENABLE_TELEMETRY", true,
+	EnableTelemetry = env.RegisterBoolVar("PILOT_ENABLE_TELEMETRY_LABEL", true,
```
Comment (bianpengyuan) [inline pilot/pkg/features/pilot.go:403]: Let's change the variable name as well.

Comment (ramaraochavali) [reply pilot/pkg/features/pilot.go]: sorry. Done now.

**Review by bianpengyuan — APPROVED** [review]

**Comment by ramaraochavali** [comment]
@hzxuzhonghu @howardjohn can you PTAL?

**Review by howardjohn — APPROVED** [review]

**Pull request merged**